src/main/java/com/twilio/oai/resolver/java/JavaConventionResolver.java

```diff
@@ -105,12 +117,32 @@ public CodegenProperty resolveEnumProperty(CodegenProperty property, String reso
                 if (resourceName != null) {
                     // It will restrict the data type to be ResourceName.ResourceName.EnumName.
                     property.dataType = property.dataType != null && property.dataType.contains(resourceName + ApplicationConstants.DOT)
-                            ? property.dataType: resourceName + ApplicationConstants.DOT + property.dataType;
+                            ? property.dataType : resourceName + ApplicationConstants.DOT + property.enumName;
                 }
 
             }
             return property;
         }
         return property;
     }
+
+    public Map<String, Object> populateSecurityAttributes(CodegenOperation co) {
```

Comment:
Is is possible to have all these in a map and then add to venderextension.
Reason: Purpose of all these extensions is to choose package/classname in case of different rest client.

Reply:
yes, added as a map